### PR TITLE
[PPD-185] strategia rilasci versioning 

### PR DESCRIPTION
**Pull request by jacopocarlini**
#### List of Changes

Added a Github action.
on merge event the action gets the labels of the PR and call the Azure pipeline to bump the version and deploy in DEV.

 - `breaking-change` -> major
 - `enhancement` -> minor
 - `bug` -> patch
 - `skip-release` -> no bump version

Use one of the tag between `gpd`, `payments`, `reporting-service`, `reporting-batch` and `reporting-analysis` to specify the component to deploy

Note: To use the github action it needed an [access token ](https://docs.microsoft.com/it-it/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows).  ⚠️ **This token is personal and it expires.**

#### Motivation and Context

to automate the deploying phase

#### How Has This Been Tested?

manually

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    if: ${{ github.event.pull_request.merged }}
+    name: Call Azure Build Pipeline
```
Comment (pasqualespica) [inline .github/workflows/deploy.yml:18]: call it `Call Azure Build Pipeline to Build and Deploy`

File: .github/workflows/sonar_analysis.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    branches:
+      - main
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    name: Call Azure Build Pipeline
```
Comment (pasqualespica) [inline .github/workflows/sonar_analysis.yml:15]: call it `Call Azure Build Pipeline to SonarCloud check`

File: .github/workflows/update_infra.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    if: ${{ github.event.pull_request.merged }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # set repo
+      - name: Set GPD repo
+        run: |
+          echo "REPO=gpd" >> $GITHUB_ENV
```
Comment (pasqualespica) [inline .github/workflows/update_infra.yml:27]: correct ?
image

File: .github/workflows/update_infra.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'reporting-analysis') }}
+
+
+      # prepare openapi template for infra repo
+      - run: |
+          mkdir -p "${GITHUB_WORKSPACE}/infra"
+          jq  '."servers"[0]."url" |= "${host}"' "${GITHUB_WORKSPACE}/${{env.REPO}}/openapi/openapi.json" > "${GITHUB_WORKSPACE}/infra/_openapi.json.tpl"
+
+      # open a PR on infra repo
+      - name: Create pull request
```
Comment (pasqualespica) [inline .github/workflows/update_infra.yml:50]: ok for the moment, next step should be useful open PR iif file is changed

**Review by pasqualespica — CHANGES_REQUESTED** [review]
great work 💯

File: .github/workflows/assignee.yml [hunk 5]
```diff
@@ -4,7 +4,9 @@ name: Auto Assign
 on:
   # Triggers the workflow on push or pull request events but only for the main branch
   pull_request:
-    branches: [ main ]
+    branches:
+      - main
+    types: [ opened ]
```
[the remark this answers is not in the document]
Comment (jacopocarlini) [reply .github/workflows/assignee.yml]: it starts only on opened/reopened event

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'reporting-batch') }}
+
+      - name: Set Reporting Service repo
+        run: |
+          echo "REPO=reporting-service" >> $GITHUB_ENV
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'reporting-service') }}
+
+
+      - name: Azure Pipelines Action - Jversion
+        uses: jacopocarlini/azure-pipelines@v1.3
```
[the remark this answers is not in the document]
Comment (jacopocarlini) [reply .github/workflows/deploy.yml:78]: ok, added a task in the backlog -> https://pagopa.atlassian.net/browse/PPD-195

File: .github/workflows/sonar_analysis.yml (hunk 2 above)
Comment (jacopocarlini) [reply .github/workflows/sonar_analysis.yml:15]: -> Call Azure **Code Review** Pipeline to SonarCloud check

File: .github/workflows/update_infra.yml (hunk 4 above)
Comment (jacopocarlini) [reply .github/workflows/update_infra.yml:50]: there is [this condition ]( https://github.com/pagopa/pagopa-debt-position/pull/103/files/5370272406db6d5f4641d1e8600a6e830539b340#diff-214de57818e42135e36dd46f4416645f6bb84ab4afb3f789913b2c6f8ef798d6R8-R11)yet

File: .github/workflows/update_infra.yml (hunk 3 above)
Comment (jacopocarlini) [reply .github/workflows/update_infra.yml:27]: yes, for the moment

File: .devops/deploy-pipelines.yml [hunk 7]
```diff
@@ -63,6 +63,11 @@ variables:
     DATASOURCE_PASSWORD: $(PROD_POSTGRES_DATASOURCE_PASSWORD)
     DATASOURCE_URL: $(PROD_POSTGRES_DATASOURCE_URL)
 
+  ${{ if eq(variables['Build.SourceBranchName'], 'merge') }}:
```
[the remark this answers is not in the document]
Comment (jacopocarlini) [reply .devops/deploy-pipelines.yml:67]: done

File: .github/workflows/update_infra.yml (hunk 4 above)
Comment (pasqualespica) [reply .github/workflows/update_infra.yml:50]: right , sorry

**Review by pasqualespica — APPROVED** [review]

**Pull request merged**